lib/url.js

```diff
@@ -353,6 +353,13 @@ function urlFormat(obj) {
   // this way, you can call url_format() on strings
   // to clean up potentially wonky urls.
   if (typeof obj === 'string') obj = urlParse(obj);
+
+  if (typeof obj !== 'object')
```

Comment:
I'd recommend saving `typeof obj` above. There are several repetitions. The ternary on the error looks like a fine solution otherwise.